### DISCUSSION TITLE
The resources declared by a Control (or a base class of the Control) can cause freezing when there are multiple references to them.

### DISCUSSION THREAD

**Discussion by NeverMorewd** [AvaloniaUI/Avalonia#15021] [Q&A] [answered]
### Describe the bug

The following codes for reference.

```
  <Button x:Key="ButtonTest" Content="ButtonHaha" />`

  <StackPanel>
     <Button Content="{StaticResource ButtonTest}" />
     <Button Content="{StaticResource ButtonTest}" />
     <Button Content="{StaticResource ButtonTest}" />
     <Button Content="{StaticResource ButtonTest}" />
 </StackPanel>
```

### To Reproduce

Compile and run

### Expected behavior

Avoid freezing or getting stuck and

1. Report the error during the compilation.
or
2. Provider extra property or attribute to support this case.
    Refer to wpf: https://learn.microsoft.com/en-us/dotnet/desktop/xaml-services/xshared-attribute
or
3. Support directly.

### Avalonia version

11.0.0

### OS

Windows

**Comment by maxkatz6** [comment, marked as answer]
This code must throw an exception, since you can't attach the same control to multiple parents.

Also, Avalonia already supports `<Template>`:
```
        <Template>
          <Button Content="ButtonHaha" />`
        </Template>
```
Where single instance of Template can be reused.

**Reply by NeverMorewd** [reply]
All right. There is compilaton error in Visual Studio with 11.1.0-beta1

![image](https://github.com/AvaloniaUI/Avalonia/assets/10544200/133365ef-e63d-4321-9df5-372252488ff0)
